File: 0023-merge-k-sorted-lists/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeKLists(ListNode[] lists) {
        int min = (int)(1e9);
        int index = -1;
        for(int i=0; i<lists.length; i++){
            if(lists[i]!=null && lists[i].val<min){
                min = lists[i].val;
                index = i;
            }
        }
        if(index==-1){
            return null;
        }
        ListNode startNode = lists[index];
        lists[index] = lists[index].next;
        startNode.next = mergeKLists(lists);
        return startNode;
        
    }
}

File: 0023-merge-k-sorted-lists/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeKLists(vector<ListNode*>& lists) {
        int min = (int)(1e9);
        int index = -1;
        for(int i = 0; i<lists.size(); i++){
            if(lists[i]!=NULL && lists[i]->val<min){
                min = lists[i]->val;
                index = i;
            }
        }
        if(index==-1){
            return NULL;
        }
        ListNode* startNode = lists[index];
        lists[index] = lists[index]->next;
        startNode->next = mergeKLists(lists);
        return startNode;
        
    }
};
